bottlecap/src/traces/trace_flusher.rs

```diff
@@ -177,17 +182,31 @@ impl TraceFlusher for ServerlessTraceFlusher {
             }));
         }
 
+        let mut join_set = JoinSet::new();
         for task in tasks {
-            match task.await {
-                Ok(result) => {
-                    if let Err(e) = result {
-                        error!("Error sending trace: {e:?}");
-                        // Return the original traces for retry
-                        return Some(coalesced_traces.clone());
+            join_set.spawn(task);
+        }
+
+        while let Some(result) = join_set.join_next().await {
+            match result {
+                Ok(inner_result) => {
+                    match inner_result {
+                        Ok(send_result) => {
+                            if let Err(e) = send_result {
+                                error!("Error sending trace: {e:?}");
+                                // Return the original traces for retry
+                                return Some(coalesced_traces.clone());
+                            }
+                        }
+                        Err(e) => {
+                            error!("Task join error: {e:?}");
+                            // Return the original traces for retry if a task panics
+                            return Some(coalesced_traces.clone());
+                        }
```

Comment:
Feel like this could be done in a cleaner way

Maybe something like:

```
while let Some(result) = join_set.join_next().await {
    let success = result
        .map_err(|e| error!("JoinSet error: {e:?}"))
        .and_then(|inner| inner.map_err(|e| error!("Task join error: {e:?}")))
        .and_then(|send_result| send_result.map_err(|e| error!("Error sending trace: {e:?}")))
        .is_ok();
    
    if !success {
        return Some(coalesced_traces.clone());
    }
}
```

Reply:
Yeah this was rough